File: exceptions-classes-AdamWalters1/src/program1/program1.py
if __name__ == "__main__":
    str_invalid_input = "Invalid input. You must enter numbers."
    
    try:
        # TODO: Add code here
        num1 = float(input("Enter first number: "))
        num2 = float(input("Enter seconds number: "))
        opr = input("Enter an operator (+,-,*,/): ")
        
        print(str(num1)+" "+opr+" "+str(num2)+" = ")
        
        
            
        while True:
            try:
                if opr not in ["+","-","*","/"]:
                    raise ValueError
                result = eval(f"{num1}{opr}{num2}")
                break
            except ZeroDivisionError as e:
                print(e)
                num2 = float(input("Enter the divisor again"))
                
        print(result)
    except Exception:
        print(str_invalid_input)
        print("Program Terminates")
           
                

File: exceptions-classes-AdamWalters1/src/program1/program1.cpp
#include <iostream>
#include <iomanip>
#include <string>

using namespace std;

int main()
{
	float num1,num2;
	char opr;
	bool done = false;
	string str = "Invalid Input. Your must enter numbers. ";
	cout << fixed << showpoint << setprecision(2);

	try{
		cout << "Enter two numbers: ";
		cin >> num1 >> num2;
		cout << endl;

		if(!cin)
			throw str;
		cout << "Enter operator (+,-,*,/): ";
		cin >> opr;
		cout << endl;

		cout << num1 << " " << opr << num2 << " = ";

		switch(opr){
			case '+':
				cout << num1+num2 << endl;
				break;
			case '-':
				cout << num1-num2 << endl;
				break;
			case '*':
				cout << num1*num2 << endl;
				break;
			case '/':
				do{
					try{
						if(num2==0)
							throw runtime_error("Attempted to divide by 0");
						cout << num1/num2 << endl;
						done = true;
					}
					catch(const exception& e){
						cerr << e.what();
						cin.clear();
						cin.ignore(100,'\n');
						cout << "Enter in the divisor again: ";
						cin >> num2;
						cout << endl;
						cout << num1 <<" " << opr << " " << num2 << " = ";

					}
				}while(!done);
				break;
			default:
				cout << "Illegal operation entered" << endl;
		}
	}
	catch(string messageStr){
		cout << messageStr << endl;
		cout << "Program terminates!" << endl;
	}
	return 0;
}
